src/sagemaker/jumpstart/factory/estimator.py

```diff
@@ -0,0 +1,338 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module stores JumpStart Estimator factory methods."""
+from __future__ import absolute_import
+
+
+from typing import Any, Dict, List, Optional
+from sagemaker import (
+    hyperparameters as hyperparameters_utils,
+    image_uris,
+    instance_types,
+    metric_definitions as metric_definitions_utils,
+    model_uris,
+    script_uris,
+)
+from sagemaker.jumpstart.artifacts import _retrieve_kwargs
+from sagemaker.jumpstart.constants import (
+    JUMPSTART_DEFAULT_REGION_NAME,
+    TRAINING_ENTRY_POINT_SCRIPT_NAME,
+)
+from sagemaker.jumpstart.enums import JumpStartScriptScope, KwargUseCase
+from sagemaker.jumpstart.factory import model
+from sagemaker.jumpstart.types import (
+    JumpStartEstimatorDeployKwargs,
+    JumpStartEstimatorFitKwargs,
+    JumpStartEstimatorInitKwargs,
+    JumpStartKwargs,
+)
+from sagemaker.jumpstart.utils import update_dict_if_key_not_present
+from sagemaker.predictor import Predictor
+
+
+def get_init_kwargs(
+    model_id: str,
+    model_version: Optional[str] = None,
+    instance_type: Optional[str] = None,
+    instance_count: Optional[int] = None,
+    region: Optional[str] = None,
+    image_uri: Optional[str] = None,
+    model_uri: Optional[str] = None,
+    source_dir: Optional[str] = None,
+    entry_point: Optional[str] = None,
+    hyperparameters: Optional[dict] = None,
+    metric_definitions: Optional[List[dict]] = None,
+    kwargs: Optional[dict] = None,
+) -> JumpStartEstimatorInitKwargs:
```

Comment:
same comment as for `Model` constructor:
- ensure you are only arguments that exists in parent [constructor](https://github.com/aws/sagemaker-python-sdk/blob/409546d47eaa642f9598d0bbf966ea302e87f638/src/sagemaker/estimator.py#L2376)
- expose at a minimum
    - `role`
    - `volume_size`
    - `volume_kms_key`
    -  `output_path`
    - `output_kms_key`
    - `sagemaker_session`
    - `tags`
    - `subnets`
    - `security_group_ids`
    - `encrypt_inter_container_traffic`
    - `enable_network_isolation`

Please also consider making `model_id` optional so that we can extend to `task`